<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CAD</class>
 <widget class="QMainWindow" name="CAD">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>936</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>CAD</string>
  </property>
  <property name="windowIcon">
   <iconset resource="cad_res.qrc">
    <normaloff>:/img/res/shape.png</normaloff>:/img/res/shape.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QTreeView" name="treeView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>256</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>512</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="statusTip">
        <string>就绪</string>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="headerHidden">
        <bool>true</bool>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
      <widget class="ClientWidget" name="client_widget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="statusTip">
        <string>就绪</string>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>936</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>绘制</string>
    </property>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>图形</string>
     </property>
     <addaction name="action_line"/>
     <addaction name="action_rectangele"/>
     <addaction name="action_ellipse"/>
    </widget>
    <addaction name="menu_2"/>
    <addaction name="action_draw_setting"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>关于</string>
    </property>
    <addaction name="action1_aboutQT"/>
    <addaction name="action_aboutme"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="action_select"/>
    <addaction name="separator"/>
    <addaction name="action_rotation"/>
    <addaction name="action_undo"/>
    <addaction name="action_recover"/>
    <addaction name="separator"/>
    <addaction name="action_delete"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="action_file_open"/>
    <addaction name="action_file_save"/>
    <addaction name="separator"/>
    <addaction name="action_quit"/>
   </widget>
   <addaction name="menu_5"/>
   <addaction name="menu_4"/>
   <addaction name="menu"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="statusTip">
    <string>就绪</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_file_open"/>
   <addaction name="action_file_save"/>
   <addaction name="separator"/>
   <addaction name="action_line"/>
   <addaction name="action_rectangele"/>
   <addaction name="action_ellipse"/>
   <addaction name="separator"/>
   <addaction name="action_draw_setting"/>
   <addaction name="separator"/>
   <addaction name="action_select"/>
   <addaction name="action_rotation"/>
   <addaction name="action_undo"/>
   <addaction name="action_recover"/>
   <addaction name="separator"/>
   <addaction name="action_delete"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_draw_setting">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/setting.png</normaloff>:/img/res/setting.png</iconset>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
   <property name="statusTip">
    <string>绘制设置 </string>
   </property>
  </action>
  <action name="action_line">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/line.png</normaloff>:/img/res/line.png</iconset>
   </property>
   <property name="text">
    <string>直线</string>
   </property>
   <property name="toolTip">
    <string>直线</string>
   </property>
   <property name="statusTip">
    <string>画直线</string>
   </property>
  </action>
  <action name="action_rectangele">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/rect.png</normaloff>:/img/res/rect.png</iconset>
   </property>
   <property name="text">
    <string>矩形</string>
   </property>
   <property name="toolTip">
    <string>矩形</string>
   </property>
   <property name="statusTip">
    <string>画矩形</string>
   </property>
  </action>
  <action name="action_ellipse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/ellipse.png</normaloff>:/img/res/ellipse.png</iconset>
   </property>
   <property name="text">
    <string>圆</string>
   </property>
   <property name="toolTip">
    <string>圆</string>
   </property>
   <property name="statusTip">
    <string>画园</string>
   </property>
  </action>
  <action name="action1_aboutQT">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/support.png</normaloff>:/img/res/support.png</iconset>
   </property>
   <property name="text">
    <string>关于QT</string>
   </property>
   <property name="toolTip">
    <string>关于QT</string>
   </property>
   <property name="statusTip">
    <string>关于QT</string>
   </property>
  </action>
  <action name="action_select">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/selection.png</normaloff>:/img/res/selection.png</iconset>
   </property>
   <property name="text">
    <string>选择</string>
   </property>
   <property name="statusTip">
    <string>选择图形</string>
   </property>
  </action>
  <action name="action_delete">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/delete.png</normaloff>:/img/res/delete.png</iconset>
   </property>
   <property name="text">
    <string>删除</string>
   </property>
   <property name="statusTip">
    <string>删除图形</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_rotation">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/rotat_right.png</normaloff>:/img/res/rotat_right.png</iconset>
   </property>
   <property name="text">
    <string>旋转</string>
   </property>
   <property name="statusTip">
    <string>旋转图形</string>
   </property>
  </action>
  <action name="action_undo">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/undo.png</normaloff>:/img/res/undo.png</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="statusTip">
    <string>撤销操作</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="action_recover">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/recover.png</normaloff>:/img/res/recover.png</iconset>
   </property>
   <property name="text">
    <string>恢复</string>
   </property>
   <property name="statusTip">
    <string>恢复操作</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="action_file_open">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/file1.png</normaloff>:/img/res/file1.png</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="statusTip">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_file_save">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/save1.png</normaloff>:/img/res/save1.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="statusTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_quit">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/logout.png</normaloff>:/img/res/logout.png</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="statusTip">
    <string>退出</string>
   </property>
  </action>
  <action name="action_aboutme">
   <property name="icon">
    <iconset resource="cad_res.qrc">
     <normaloff>:/img/res/shape.png</normaloff>:/img/res/shape.png</iconset>
   </property>
   <property name="text">
    <string>关于我</string>
   </property>
   <property name="toolTip">
    <string>关于我</string>
   </property>
   <property name="statusTip">
    <string>关于我</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ClientWidget</class>
   <extends>QWidget</extends>
   <header>ClientWidget.h</header>
   <container>1</container>
   <slots>
    <signal>on_tree_view_add_information()</signal>
    <signal>on_tree_view_update_checked()</signal>
    <signal>update_tree_view_item()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="cad_res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>client_widget</sender>
   <signal>on_tree_view_add_information()</signal>
   <receiver>CAD</receiver>
   <slot>on_tree_view_add_information()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>601</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>client_widget</sender>
   <signal>on_tree_view_update_checked()</signal>
   <receiver>CAD</receiver>
   <slot>on_tree_view_update_checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>683</x>
     <y>-11</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>client_widget</sender>
   <signal>update_tree_view_item()</signal>
   <receiver>CAD</receiver>
   <slot>update_tree_view_item()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>477</x>
     <y>-2</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_tree_view_add_information()</slot>
  <slot>update_tree_view_item()</slot>
  <slot>on_tree_view_update_checked()</slot>
 </slots>
</ui>
